tools/astarte_export/lib/astarte/fetchdata/fetchdata.ex

```diff
@@ -285,9 +285,42 @@ defmodule Astarte.Export.FetchData do
     end
   end
 
+  defp from_native_type(value, :binaryblobarray) when is_list(value) do
+    case value do
+      list when is_list(list) -> Enum.map(list, fn x -> from_native_type(x, :binaryblob) end)
+      _ -> value
+    end
+  end
+
   defp from_native_type(value, :binaryblob) do
-    {:ok, binary_blob} = Base.encode64(value)
-    binary_blob
+    case value do
+      <<_::binary>> -> Base.encode64(value)
+      _ -> value
```

Comment:
You could reduce this to a `is_binary(value)` check in the function head. If the value is not a binary, [let it crash](https://erlang.org/pipermail/erlang-questions/2003-March/007870.html)! This way, you'll be also keeping the same behavior as before